modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -139,6 +221,31 @@ typename Net::ResultL infer(cv::GArray<cv::Rect> roi, Args&&... args) {
     return GInferList<Net>::on(roi, std::forward<Args>(args)...);
 }
 
+/** @brief Calculates responses for the specified network (template
+ *     parameter) for every region in the source image, extended version.
+ *
+ * @tparam A network type defined with G_API_NET() macro.
+ * @param image A source image containing regions of interest
+ * @param args GArray<> objects of cv::Rect or cv::GMat, one per every
+ * network input:
+ * - If a cv::GArray<cv::Rect> is passed, the appropriate
+ *   regions are taken from `image` and preprocessed to this particular
+ *   network input;
+ * - If a cv::GArray<cv::GMat> is passed, the underlying image data is
+ *   preprocessed to this input; if the GMat is a tensor, it is passed
+ *   as-is.
```

Comment:
It seems I've missed this functionality, this is what needs to be added and tested.

Reply:
OR: forbidden for now (comment to be changed) with a follow-up task in our tracker.

Reply:
Decided to go with the current implementation. Not sure if a follow-up task is required either - will add once we enter such use-case. Will update the documentation header.

Reply:
Documentation header updated